server/container_create_linux.go

```diff
@@ -367,7 +367,7 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 				}
 				specgen.SetLinuxResourcesMemoryLimit(memoryLimit)
 				if resources.MemorySwapLimitInBytes != 0 {
-					if resources.MemorySwapLimitInBytes < resources.MemoryLimitInBytes {
+					if resources.MemorySwapLimitInBytes < resources.MemoryLimitInBytes && !node.CgroupIsV2() {
```

Comment:
This is not quite accurate from the spec, I think. I would expect the following logic:

- As written, this validation is correct.
- In the cgroups v1 case, set this as is currently specified on L378.
- In the cgroups v2 case, set swap to the difference between MemoryLimitInBytes and MemorySwapLimitInBytes. Since `MemoryLimitInBytes <= MemorySwapLimitInBytes`, `MemorySwapLimitInBytes - MemoryLimitInBytes >= 0`.